File: dfc_app/lib/services/service_time_service.dart
import '../providers/order_type_provider.dart';
import 'api_client.dart';

class ServiceTimeSetting {
  final String serviceType;
  final int minMinutes;
  final int maxMinutes;
  const ServiceTimeSetting({
    required this.serviceType,
    required this.minMinutes,
    required this.maxMinutes,
  });

  factory ServiceTimeSetting.fromJson(Map<String, dynamic> json) => ServiceTimeSetting(
        serviceType: json['serviceType'] as String? ?? '',
        minMinutes: json['minMinutes'] as int? ?? 0,
        maxMinutes: json['maxMinutes'] as int? ?? 0,
      );

  String get label => '$minMinutes–$maxMinutes min';
}

class ServiceTimeService {
  static Future<List<ServiceTimeSetting>> getServiceTimes() async {
    final json = await ApiClient.get('/Public/service-times') as List<dynamic>;
    return json.map((e) => ServiceTimeSetting.fromJson(e as Map<String, dynamic>)).toList();
  }

  static ServiceTimeSetting? forType(List<ServiceTimeSetting> settings, ServiceType type) {
    for (final s in settings) {
      if (s.serviceType == type.apiValue) return s;
    }
    return null;
  }
}

File: Frontend/CustomerOrderingWeb/src/app/services/service-time.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, signal } from '@angular/core';
import { environment } from '../../environments/environment';
import { ServiceTimeSetting } from '../models/service-time.model';
import { ServiceType } from './order-type.service';

@Injectable({ providedIn: 'root' })
export class ServiceTimeService {
  private baseUrl = environment.apiBaseUrl + '/Public';

  settings = signal<ServiceTimeSetting[]>([]);

  constructor(private http: HttpClient) {
    this.load();
  }

  load() {
    this.http.get<ServiceTimeSetting[]>(`${this.baseUrl}/service-times`).subscribe({
      next: (res) => this.settings.set(res),
      error: () => {}
    });
  }

  label(serviceType: ServiceType): string {
    const setting = this.settings().find(s => s.serviceType === serviceType);
    if (!setting) return '';
    return `${setting.minMinutes}–${setting.maxMinutes} min`;
  }
}
